### PR TITLE
Implement checkstyle output

### PR DESCRIPTION
**Pull request by swalberg**
Checkstyle output is used by various Java tools but most notably is well
supported by Jenkins CI. With the Checkstyle plugin, this change will
allow Jenkins to graph lint errors and provide deep links to the source.

jenkins 2016-01-22 19-55-07

Ref: #110

### CODE REVIEW

**Comment by sds** [comment]
Thanks for the pull request, @swalberg.

This from a code perspective looks great. My concern lies in the choice of third-party libraries. The [`nokogiri` gem is over 8MB](https://rubygems.org/gems/nokogiri/versions) due to its vendored libraries, so it's a pretty heavy dependency to include for such a small feature (for comparison, `haml-lint` is ~250KB). I appreciate it makes the tests easier to write, but I would prefer we avoid having to include this dependency since we're unlikely to use it for anything else.

Furthermore, the inclusion of the `builder` library is also a reasonable choice, but that project hasn't received any updates [since 2013](https://rubygems.org/gems/builder/versions/3.2.2). I'd rather not include a dependency on a library that hasn't seen _any_ development for years (there are open pull requests from 2014).

Given how simple the Checkstyle XML schema is, and given how unlikely it is (in my mind) that `haml-lint` will use an XML parsing library for anything other than this formatter, I'd prefer it if this formatter was implemented with simple string concatenation, for example what was done for the [`scss-lint` Checkstyle formatter](https://github.com/Sweetchuck/scss_lint_reporter_checkstyle/blob/31151505fb15f286b10987f6220208cb6e17a700/lib/scss_lint_reporter_checkstyle.rb). I know it's a bit cumbersome, but it's more in keeping with the needs of this particular feature.

Switch that over and I'll happily merge. Thanks!

**Comment by swalberg** [comment]
Fair enough. The string concatenation is easy enough and thanks for the pointers to code.

**Comment by swalberg** [comment]
Hi, very sorry about how long this took. I've made the changes and left them as a separate commit. I can squash after you've had a chance to review.

**Comment by sds** [comment]
Thanks for your work on this, @swalberg!

Merged in 86a86ec.

File: lib/haml_lint/reporter/checkstyle_reporter.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      log.log xml
+    end
+
+    private
+
+    def render_offenses(offenses)
+      xml = ''
+      offenses.each do |offense|
+        xml << "<error line=\"#{offense.line}\" severity=\"#{offense.severity}\" "
+        xml << "message=\"#{offense.message}\" source=\"#{offense.linter.name}\" />"
```
Comment (oliverklee) [inline lib/haml_lint/reporter/checkstyle_reporter.rb]: Won't this break the XML if offense.message contains a double quote?

Comment (swalberg) [inline lib/haml_lint/reporter/checkstyle_reporter.rb]: Almost certainly, good catch. Do you think a `gsub` will be enough to add an escape, or are there some other chars that I'm missing?

Comment (oliverklee) [inline lib/haml_lint/reporter/checkstyle_reporter.rb]: CGI::escapeHTML seems to be the recommended way to do so:
https://stackoverflow.com/questions/692921/how-to-html-encode-escape-a-string-is-there-a-built-in

**Pull request closed without merging**